libbpfgo.go

```diff
@@ -841,6 +861,14 @@ func (b *BPFMap) Update(key, value unsafe.Pointer) error {
 	return nil
 }
 
+func (b *BPFMap) UpdateFlags(key, value unsafe.Pointer, flags MapFlag) error {
```

Comment:
The function name makes it look like it is updating the "flags" for a map, while actually we're just updating the map using flags argument. Should we change the method name ?

I think I'm ok with not breaking the API to have flags added, at least until we decide to break the API and unify those 2 (should we add a TODO at the code ?). Also, we could make Update() be a call to UpdateFlags() (with a new name) with a nil flag argument (something like libbpf does when deprecating functions).